src/registrar/migrations/0122_delete_cascade_submitter_contacts.py

```diff
@@ -0,0 +1,27 @@
+from django.db import migrations
+from django.db.models import Q
+from typing import Any
+
+
+# Deletes Contact objects associated with a submitter which we are deprecating
+def cascade_delete_submitter_contacts(apps, schema_editor) -> Any:
+    contacts_model = apps.get_model("registrar", "Contact")
+    submitter_contacts = contacts_model.objects.filter(
+        Q(submitted_domain_requests__isnull=False) | Q(submitted_domain_requests_information__isnull=False)
+    ).filter(
+        information_senior_official__isnull=True,
+        senior_official__isnull=True,
```

Comment:
why are you checking for SO here?

Reply:
We're deleting all submitter Contacts from our data but we don't want to delete a submitter contact if they're a SO